cpg_infra/abstraction/group_settings.py

```diff
@@ -0,0 +1,113 @@
+"""
+Contains pulumi.dynamic.ResourceProvider implementations
+for Google Groups settings.
+"""
+
+from functools import cache
+import pulumi
+import pulumi.dynamic
+import google.auth
+import googleapiclient.discovery
+from google.auth.transport.requests import Request
+
+
+class GroupSettings(pulumi.dynamic.Resource):
+    """A Pulumi dynamic resource for Google Groups settings."""
+
+    group_email: pulumi.Output[str]
+    settings: pulumi.Output[dict]
+
+    def __init__(
+        self,
+        name: str,
+        group_email: str,
+        settings: dict,
+        opts: pulumi.ResourceOptions | None = None,
+    ):
+        super().__init__(
+            GroupSettingsProvider(),
+            name,
+            {
+                'group_email': pulumi.Output.from_input(group_email),
```

Comment:
This is surprising Pulumi can't figure it out automatically

Reply:
Thanks, it actually can -- it's a bit more concise now.